plugins/minimator/src/minimator.cpp

```diff
@@ -185,7 +129,151 @@ class MinimatorLaneSensor : public cloe::LaneBoundarySensor {
  private:
   cloe::LaneBoundaries lane_boundaries_;
   cloe::Frustum frustum_;
-  Eigen::Isometry3d mount_pose_ = Eigen::Isometry3d::Identity();
+  Eigen::Isometry3d mount_pose_;
+};
+
+/**
+ * The SimulatorSensorMockup class provides dummy values for the World sensor and Ego sensor.
+ * The Ego Sensor values are provided with respect to the Cloe frame (center of the
+ * rear axle).
+ * The world sensor values are provided relative to the sensor mounting position.
+*/
+class SimulatorSensorMockup {
+ public:
+  SimulatorSensorMockup(const uint16_t& id, const SensorMockupConfig& sensor_mockup_config)
+      : vehicle_id(id), sensor_mockup_config_(sensor_mockup_config) {}
+
+  void process_ego_vehicles() {
+    std::array<double, 3> ori{0.0, 0.0, 0.0};
+    std::array<double, 3> pos{sensor_mockup_config_.ego_sensor_mockup.ego_object.position.x,
+                              sensor_mockup_config_.ego_sensor_mockup.ego_object.position.y,
+                              sensor_mockup_config_.ego_sensor_mockup.ego_object.position.z};
+    if (ego_objects_.empty()) {
+      auto o = std::make_shared<cloe::Object>();
+      o->id = this->vehicle_id;
+      o->dimensions = Eigen::Vector3d(5.0, 2.00, 1.8);
+      Eigen::Quaterniond quaternion = cloe::utility::quaternion_from_rpy(ori[0], ori[1], ori[2]);
+      Eigen::Vector3d translation = Eigen::Vector3d(pos[0], pos[1], pos[2]);
+      o->pose = cloe::utility::pose_from_rotation_translation(quaternion, translation);
+      o->velocity =
+          Eigen::Vector3d(sensor_mockup_config_.ego_sensor_mockup.ego_object.velocity, 0.0, 0.0);
+      ego_objects_.emplace_back(std::move(o));
+    }
+  }
+
+  void process_sensed_objects() {
+    if (world_objects_.empty()) {
+      for (const auto& object : sensor_mockup_config_.object_sensor_mockup.objects) {
+        std::array<double, 3> ori{0.0, 0.0, 0.0};
+        std::array<double, 3> pos{object.position.x, object.position.y, object.position.z};
+        auto o = std::make_shared<cloe::Object>();
+        o->id = this->vehicle_id;
+        o->dimensions = Eigen::Vector3d(5.0, 2.00, 1.8);
+        Eigen::Quaterniond quaternion = cloe::utility::quaternion_from_rpy(ori[0], ori[1], ori[2]);
+        Eigen::Vector3d translation = Eigen::Vector3d(pos[0], pos[1], pos[2]);
+        o->pose = cloe::utility::pose_from_rotation_translation(quaternion, translation);
+        // User should extend the logic to calculate the position of the object if they want to have a
+        // velocity different than zero.
+        UnsupportedValueException(
```

Comment:
This should inherit from `cloe::ModelError`, otherwise cloe-engine won't be able to terminate cleanly.

Reply:
I modified the code. Looks like the condition to throw was also missing :/. Thank you for catching this!